### ISSUE TITLE
rumdl doesn't respect `line_length`

### ISSUE THREAD

**Issue by arv-anshul** [rvben/rumdl#528]
### rumdl version

0.1.47

### Installation method

Pre-built binary

### Bug description

`rumdl` doesn't respect `line_length` for **reference-style link definition**.

### Steps to reproduce

Format the below code block:

```md
[polars.expr.qcut]: https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.qcut.html "Bin continuous values into discrete categories based on their quantiles."
```

### Expected behavior

Should not complain because this is valid.

### Actual behavior

```
temp.md:1:121: [MD013] Line length 181 exceeds 120 characters
```

### Configuration

```toml
[global]
respect-gitignore = true
flavor = "standard"
line_length = 120
include = [
  "content/**/*.{md,mdx}",
  "src/**/*.{md,mdx}",
  "README.md",
]
disable = [
  "MD036", # no-emphasis-as-heading
  "MD026", # no-trailing-punctuation
]

[per-file-flavor]
"**/*.md" = "gfm"
"**/*.mdx" = "mdx"

[MD033]
allowed-elements = ["br", "details", "summary", "figure", "figcaption", "iframe", "kbd", "a", "img"]

[MD013]
reflow = true
code-blocks = false
headings = false
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.48. Link reference definitions with titles are now exempt from MD013 line length warnings, since they cannot be meaningfully split across lines. Thanks for reporting, @arv-anshul!
